packages/edit-site/src/components/sidebar-dataviews/index.js

```diff
@@ -34,6 +34,7 @@ export default function DataViewsSidebarContent() {
 							slug={ dataview.slug }
 							title={ dataview.title }
 							icon={ dataview.icon }
+							suffix={ <span>{ dataview.count }</span> }
```

Comment:
> Is the span needed?

Asked here: https://github.com/WordPress/gutenberg/pull/62028#discussion_r1618917031

Yes, otherwise the content will not render as a child of `<HStack />`